File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/params/set_tcp_offset.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time

h10w_params = h10w.H10wParams()
time.sleep(1)

print("=== 设置 TCP 偏移量示例 ===")
offset = []
offset.append((1, [0.1, 0.2, 0.3, 0.4, 0.5, 0.6]))
offset.append((2, [0.2, 0.3, 0.4, 0.5, 0.6, 0.7]))

ret = h10w_params.setTcpOffset(offset)
if ret != 0:
    print("设置失败")
else:
    print("设置成功")

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/params/set_tcp_offset.cpp
// set_tcp_offset.cpp

#include <iostream>
#include <iomanip>
#include "h10-w/h10w_params.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 设置 TCP 偏移量示例 ===" << std::endl;
  H10wParams params;

  // 设置 TCP 偏移量
  TcpOffsetParams tcpOffsetParams = {
      {1, {0.1, 0.2, 0.3, 0.4, 0.5, 0.6}},  // 设置左臂 TCP 偏移量参数
      {2, {0.2, 0.3, 0.4, 0.5, 0.6, 0.7}}   // 设置右臂 TCP 偏移量参数
  };

  if (!params.setTcpOffset(tcpOffsetParams)) {
    std::cout << "设置成功" << std::endl;
  } else {
    std::cout << "设置失败" << std::endl;
    return -1;
  }
  return 0;
}
